### PR TITLE
Rep 2404 - DD Service not sending trace id

### PR DESCRIPTION
**Pull request by mknocke**
Cannot run functional tests on my computer, so opening pull request.

### CODE REVIEW

**Comment by wdschei** [comment]
ok to test

File: repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/core/tracing/TracingDistDatastoreTest.groovy [hunk 1]
```diff
@@ -96,4 +98,55 @@ class TracingDistDatastoreTest extends ReposeValveTest {
         lines.size() == 1
     }
 
+    def "DistDatastore service request should be added the tracing header"() {
```
Comment (adrianjgeorge) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/core/tracing/TracingDistDatastoreTest.groovy]: this test appears to have nothing to do with what this story is about?

File: repose-aggregator/services/datastore/impl/src/main/java/org/openrepose/core/services/datastore/impl/distributed/remote/command/AbstractRemoteCommand.java [hunk 2]
```diff
@@ -57,8 +58,9 @@ public String getBaseUrl() {
     }
 
     protected Map<String, String> getHeaders(RemoteBehavior remoteBehavior) {
-        Map<String, String> headers = new HashMap<String, String>();
+        Map<String, String> headers = new HashMap<>();
         headers.put(DatastoreHeader.HOST_KEY.toString(), hostKey);
+        headers.put(CommonHttpHeader.TRACE_GUID.toString(), remoteEndpoint.getHostName());
```
Comment (adrianjgeorge) [inline repose-aggregator/services/datastore/impl/src/main/java/org/openrepose/core/services/datastore/impl/distributed/remote/command/AbstractRemoteCommand.java]: you should be using the same guid that is currently being run under on this node

**Comment by adrianjgeorge** [comment]
Also, unlike the others you should always do this one, this is a case where we are internal within repose so we always want it

**Pull request merged**